os/os.proto

```diff
@@ -0,0 +1,324 @@
+  // the OS package and its version. This MUST be strictly informational if
+  // used, and can contain information such as build date, target platform,
+  // developer, etc.
+  string description = 2;
+}
+
+// The InstallError message MUST be sent by the Target to the Client whenever an
+// issue occurs. The Target MUST immediately close the RPC without a gRPC error.
+message InstallError {
+  enum Type {
+    // An unspecified error. Must use the detail value to describe the issue.
+    UNSPECIFIED = 0;
+    // The newly transferred package is not compatible with the Target platform.
+    // The detail field MUST contain the detailed error message.
+    INCOMPATIBLE = 1;
+    // The OS package being transferred is larger than the available size the
+    // Target provisioned. This is unexpected since the Target MUST clear disk
+    // space for the new OS packages. The available space and the OS package
+    // size MUST be guaranteed by the platform maker, therefore the most likely
+    // cause of this error is that a wrong package is being transferred.
+    TOO_LARGE = 2;
+    // Used whenever the system is unable to parse the newly transferred
+    // package, like reading the OS version or the integrity checksums.
+    PARSE_FAIL = 3;
+    // The transferred OS package fails integrity check.
+    INTEGRITY_FAIL = 4;
+    // Attempting to force transfer an OS package with the same version as the
+    // currently running.
+    INSTALL_RUN_PACKAGE = 5;
+    // Another Install RPC to this Target is already in progress.
+    INSTALL_IN_PROGRESS = 6;
+    // A switchover happened during the Install RPC.
+    UNEXPECTED_SWITCHOVER = 7;
+    // Failed to sync the transferred OS package to the standby Supervisor. The
+    // detail value MUST have more information.
+    SYNC_FAIL = 8;
+  }
+  Type type = 1;
+  string detail = 2;
+}
+
+// The ActivateRequest is sent by the Client to the Target to initiate a change
+// in the next bootable OS version that is to be used on the Target.
+message ActivateRequest {
+  // The version that is required to be activated and booted.
+  string version = 1;
+  // For dual Supervisors setting this flag instructs the Target to perform the
+  // action on the Standby Supervisor.
+  bool standby_supervisor = 2;
+}
+
+// The ActivateResponse is sent from the Target to the Client in response to the
+// Activate RPC. It indicates the success of making the OS package version
+// active.
+message ActivateResponse {
+  oneof response {
+    ActivateOK activate_ok = 1;
+    ActivateError activate_error = 2;
+  }
+}
+
+// If the Target is already running the requested version in ActivateRequest,
+// then it replies with ActivateOK. If the Target has the OS package version
+// requested in ActivateRequest then it replies with ActivateOK and proceeds to
+// boot. In a Target with dual Supervisor, performing this RPC on the Active
+// Supervisor triggers a switchover before booting the (old)Active Supervisor.
```

Comment:
So with a dual supervisor, the requirement is to do an ISSU upgrade?

Reply:
I believe ISSU upgrade is a bit more involved. This is just stating that if standby_supervisor = False, Then the Active Supervisor is the Target of the package activation, regardless of the active package in the Standby Supervisor. Because the Active Supervisor is rebooting, a switchover needs to happen.

Reply:
Thanks Sam. So the image activation specified here doesn't require ISSU upgrade?

Reply:
The platform should always attempt performing a failover with the least impact possible. I will update the text.

Reply:
I know this does not answer your question. The answer is out of of scope of this service. The switchover behavior in regards to non stop forwarding should be a customer to vendor conversation. ISSU is rather specific and packages upgrade and switchover behavior together.